content/source/docs/enterprise/api/oauth-clients.html.md

```diff
@@ -26,10 +26,12 @@ This endpoint allows you to create a VCS connection between an organization and
 
 ~> **Note:** This endpoint does not currently support creation of a Bitbucket Server OAuth Client.
 
-Result  | Status and response
---------|------------------------
-Success | HTTP 201 and a JSON API document (`type: "oauth-clients"`).
-Failure | HTTP 4XX and a JSON API list of error messages.
+Status  | Response                                | Reason
+--------|-----------------------------------------|----------
+[201][] | [JSON API document][] (`type: "oauth-clients"`) | The request was successful
+[404][] | [JSON API error object][]               | Organization not found or user unauthorized to perform action
+[422][] | [JSON API error object][]               | Malformed request body (missing attributes, wrong types, etc.)
+[500][] | [JSON API error object][]               | Failure during OAuth Client creation
```

Comment:
Hm...we typically don't document 500s as return codes, because we don't want 500s to ever be user-facing ideally (although that's aspirational :). What I was suggesting here was that if different codes represent distinguishable issues in the creation (which is the case for e.g. the Admin SMTP settings) then those should all be listed. Otherwise, if all the cases where it would fail due to user error are covered by 422, then we can just omit 500 and stick with 404 and 422.

Reply:
:+1: 404 and 422 should cover all missing or invalid attributes.

The 500 error has been removed. It was copied from the template (`content/source/docs/enterprise/api/_template.md`), so we may want to update that document.